infer/crates/kanalizer-rs/src/inference.rs

```diff
@@ -406,7 +425,19 @@ impl Kanalizer {
             self.validate_input(input)?;
         }
         let input = input.chars().map(|c| c.to_string()).collect::<Vec<_>>();
-        Ok(self.inner.infer(&input, options).into_iter().collect())
+        let infer_result = self.inner.infer(&input, options);
+        let output = infer_result
+            .output
+            .iter()
+            .map(|c| c.to_string())
+            .collect::<String>();
```

Comment:
[`impl<'a> FromIterator<&'a char> for String`](https://doc.rust-lang.org/stable/std/string/struct.String.html#impl-FromIterator%3C%26char%3E-for-String)

```suggestion
        let output = infer_result.output.iter().collect();
```